### PR TITLE
SNOW-1659276 sp precommit fixes

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1659276

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Comment by sfc-gh-aalam** [comment]
successful run: https://ci-dev-142.int.snowflakecomputing.com/job/PythonStoredProcBuildPrecommitTest/1337/
paired snowflake PR: https://github.com/snowflakedb/snowflake/pull/213154

File: tests/integ/test_cte.py [hunk 1]
```diff
@@ -484,6 +478,7 @@ def test_join_table_function(session):
     assert count_number_of_ctes(df_result.queries["queries"][-1]) == 1
 
 
+@pytest.mark.skipif(IS_IN_STORED_PROC, reason="Cannot create temp table in stored proc")
```
Comment (sfc-gh-aling) [inline tests/integ/test_cte.py:481]: do you think we can use this function: https://github.com/snowflakedb/snowpark-python/blob/main/tests/utils.py#L147-L148
to create table with random name in stored proc?

File: tests/integ/test_cte.py [hunk 2]
```diff
@@ -64,7 +56,7 @@ def setup(request, session):
 
 
 def check_result(session, df, expect_cte_optimized):
-    df = df.sort(sql_expr("$1"))
+    df = df.sort(df.columns)
```
Comment (sfc-gh-aling) [inline tests/integ/test_cte.py:59]: what is this change for?

Comment (sfc-gh-aalam) [reply tests/integ/test_cte.py:59]: this change was only sort first column earlier which was giving flaky results in sp environment. With this change, we are sorting all columns to ensure that all data is sorted correctly

File: tests/integ/test_cte.py (hunk 1 above)
Comment (sfc-gh-aalam) [reply tests/integ/test_cte.py:481]: giving a random name is not an issue here. We cannot create team tables using CTAS in stored procedures. We have to do it using the two queries: 
1. create scoped temp table
2. insert into table

Comment (sfc-gh-aling) [reply tests/integ/test_cte.py:481]: thanks for the context! I'm not aware of the limitation of CTAS before

Comment (sfc-gh-aling) [reply tests/integ/test_cte.py:481]: do you think we can write this? or rather we want to test this CTAS behavior.
I think rewriting should be straightforward and it helps improve test coverage.
```
table_name = Utils.random_table_name()
session.create_dataframe(
[<data>...])
.save_as_table(table_name)

**Review by sfc-gh-aling — APPROVED** [review]
pre-approve as we agree upon refactoring the test

**Comment by sfc-gh-aalam** [comment]
run after refactor: https://ci-dev-142.int.snowflakecomputing.com/job/PythonStoredProcBuildPrecommitTest/1343/

**Review by sfc-gh-jdu — APPROVED** [review]

**Pull request merged**